daemon/daemon_main.go

```diff
@@ -1338,6 +1339,14 @@ func runDaemon() {
 
 	metricsErrs := initMetrics()
 
+	if option.Config.IPAM == option.IPAMENI {
```

Comment:
I put it here because it was the easiest thing to do cognitively -- "I know that this logic must run before the API is served". And so I put it in the most obvious spot. :sweat_smile:  

If it makes sense elsewhere, happy to move it. I realize if everyone had the same logic as me, then the ordering of operations might get out of whack.